File: Source/CSharp/Expressive/Expressive/Expressions/ParenthesisedExpression.cs
using Expressive.Exceptions;
using System.Collections.Generic;

namespace Expressive.Expressions
{
    internal class ParenthesisedExpression : IExpression
    {
        private readonly IExpression innerExpression;

        internal ParenthesisedExpression(IExpression innerExpression)
        {
            this.innerExpression = innerExpression;
        }

        #region IExpression Members

        public object Evaluate(IDictionary<string, object> variables)
        {
            if (this.innerExpression is null)
            {
                throw new MissingParticipantException("Missing contents inside ().");
            }

            return this.innerExpression.Evaluate(variables);
        }

        #endregion
    }
}

File: Source/Java/Expressive/src/main/java/com/bijington/expressive/expressions/ParenthesisedExpression.java
package com.bijington.expressive.expressions;

import com.bijington.expressive.exceptions.ExpressiveException;

import java.util.Map;

/**
 * Created by shaun on 05/07/2016.
 */
public class ParenthesisedExpression implements IExpression {
    private final IExpression _innerExpression;

    public ParenthesisedExpression(IExpression innerExpression) {
        _innerExpression = innerExpression;
    }

    @Override
    public Object evaluate(Map<String, Object> variables) throws ExpressiveException {
        if (_innerExpression == null) {
            // TODO should this be an exception?
            return null;
        }

        return _innerExpression.evaluate(variables);
    }
}
